index.bs

```diff
@@ -232,27 +232,137 @@ Non-Functional Requirements {#requirements-non-functional}
+
+
+Transport and metadata discovery with QUIC {#transport}
+=======================================================
+
+If a discovering endpoint wants to connect to an advertising endpoint, or to
+learn further metdata about it, it initiates a
+[QUIC](https://tools.ietf.org/html/draft-ietf-quic-transport-16) connection to
+the IP and port from the SRV record.  Prior to authentication, some message may be
+exchanged (such as further metadata), but such info should be treated as
+unauthenticated (such as indicating to a user that a display name of an
+unauthenticated endpoint is unauthenticated).  
+
+To learn further metadata, an endpoint may send an endpoint-info-request
+message (see Appendix A) and receive back an endpoint-info-response message.  The
+messages may contain the following information with the following meaning:
+
+- display-name: The display name of the responding endpoint intended to be
+  displayed to a user by the requesting endpoint.  If the responding endpoint is
+  not yet authenticated, the requesting endpoint should make UI affordance for
+  indicating to the user that the display name is not yet authenticated.  If the
+  responding endpoint changes its display name, the requesting endpoint should
+  make UI affordance for indicating to the user that the display name has
+  changed. 
+
+- model-name: The model name of the device (if it is a device), used mainly for
+  debugging purposes, but may be displayed to the user of the requesting
+  endpoint.
+
+<!-- TODO: Add device type and/or capabilities -->
+
+If a discovering endpoint (acting as a QUIC client) wishes to receive messages
+from an advertising endpoint (acting as a QUIC server) or an advertising
+endpoint (acting as a QUIC server) wishes to send message to a discovering
+endpoint (acting as a client endpoint), it may wish to keep the QUIC connection
+alive.  Once neither side needs to keep the connection alive for the purposes of
+sending or receiving messages, the connection should be closed with an error
+code of 1.  In order to keep a QUIC connection alive, an endpoint may send an
+endpoint-status-request message, and any endpoint that receives an
+endpoint-status-request message should send an endpoint-status-response message.
+Such messages should be sent more frequently than the QUIC idle_timeout
+transport parameter (see section 18 of
+[QUIC](https://tools.ietf.org/html/draft-ietf-quic-transport-16)) and QUIC PING
+frames should not be used.  An idle_timeout transport parameter of 25 seconds is
+recommended.  If other mechanisms or traffic (such as other requests and
+responses) are serving the purpose of keeping the connection alive,
+endpoint-status-requests messages should be delayed or not used at all.
+
+If a client endpoint wishes to send messages to a server endpoint, the client
+endpoint can connect to the server endpoint "on demand"; it does not need to
+keep the connection alive.
```

Comment:
Ah, so the keepalive is always initiated from the QUIC server?

Reply:
No, the client may send them to continue receiving message from the server.